lib/hanami/cli/commands/app/routes.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require "hanami"
+require "hanami/router/inspector"
+
+module Hanami
+  module CLI
+    module Commands
+      module App
+        # Inspect the application and slice routes
+        #
+        # Without the `slice` option, it will inspect the application routes:
+        #
+        # ```
+        # $ hanami routes
+        # ```
+        #
+        # Otherwise, provide the slice name:
+        #
+        # ```
+        # $ hanami routes --slice=main
+        # ```
+        #
+        # All the formatters available from `hanami-router` are available:
+        #
+        # ```
+        # $ hanami routes --format=csv
+        # ```
+        #
+        # You can also use a custom formatter registered in the application
+        # container. You can identify it by its key:
+        #
+        # ```
+        # $ hanami routes --format=custom_routes_formatter
+        # ```
+        class Routes < Hanami::CLI::Command
+          desc "Inspect application and slice routes"
+
+          option :format,
+                 default: :human_friendly,
+                 required: false,
+                 desc: "Output format"
+
+          option :slice,
+                 required: false,
+                 desc: "Slice of the routes to print."
+
+          DEFAULT_FORMAT = "human_friendly"
+
+          # @api private
+          def call(format: DEFAULT_FORMAT, slice: nil)
+            require "hanami/prepare"
+            inspector = Hanami::Router::Inspector.new(formatter: resolve_formatter(format))
+            resolve_slice(slice).router(inspector: inspector)
+            out.puts inspector.call
+          end
+
+          private
+
+          def resolve_formatter(format)
+            resolve_formatter_from_hanami_router(format) ||
+              resolve_formatter_from_app(format)
+          end
+
+          def resolve_formatter_from_hanami_router(format)
+            file = "hanami/router/formatter/#{format}"
+            require file
+
+            file
+              .then { |f| inflector.classify(f) }
+              .then { |class_name| inflector.constantize(class_name) }
+              .then(&:new)
+          rescue LoadError
+            false
+          end
+
+          def resolve_formatter_from_app(format)
```

Comment:
@waiting-for-dev Given we're providing Human friendly and CSV, I wouldn't bother to let final users to specify a formatter.

Technical reasons are:

* One more feature to support
* We need to document where to put this file and how it should be structured. I assume that we would provide a superclass for it.
* In case of changes in on the superclass, we'll need to support a migration.

The cost/benefit analysis for me is negative: too much burden for a very little gain in terms of features.

Could you please remove it?

Reply:
Hey, @jodosha,

IMHO, that would be a beneficial feature. It'd give the option to integrate easily formatters which output could be imported into standard developer tools like Postman or OpenAPI.

> We need to document where to put this file and how it should be structured. I assume that we would provide a superclass for it.

No need for a superclass. It only needs to be [called by the inspector with the routes](https://github.com/hanami/router/blob/9c2b746d41725b3c2e511cf936db0b4ed32e76ee/lib/hanami/router/inspector.rb#L36). Yeah, of course, we need documentation, but like with everything else :slightly_smiling_face: 

> In case of changes in on the superclass, we'll need to support a migration.
No superclass. Of course, the signature of the `#call` method could change at some point. But, as long as we obey with semver, it's like any other change in Hanami. I don't see why we need to support migration.

Do you still think it's better to remove it?

Reply:
@jodosha, TBH, after all the recent changes in Hanami, I'm not sure where should it go, but it should be the standard Hanami directory for "lib" code (`lib/`, `config/providers/` ? cc @timriley)

Reply:
I haven't been following this PR and I can't catch up tonight, so all I can do at this point is answer your question based on the surrounding code. So: it looks like you want the formatter to be a component registered in the Hanami app container. In this case it should go into `app/` (or if it is a complex component that requires its own configuration before it can be registered, etc., then yes, you could make a provider for it).

Reply:
Thanks, @timriley! Thoughts, @jodosha? :slightly_smiling_face:

Reply:
After talking about it off-line, we've decided to support it in an experimental phase.